File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.java
class Solution
{	
    	public int solution(int n, int a, int b) {
		int answer = 0;// 일단 냅두자

		while (n != 0) {
			System.out.println("n : "+n + " a : "+a+" b : " + b);
			if ( (a%2==0 && a == b + 1) || (b%2==0 && b == a + 1)) {
				answer++;
				//System.out.println("조건만족");
				break;
			}
			if (a % 2 == 0) {
				a = a / 2;
			} else {
				a = (a / 2) + 1;
			}
			if (b % 2 == 0) {
				b = b / 2;
			} else {
				b = (b / 2) + 1;
			}
			answer++;
			n = n / 2;
		
		}
		return answer;
	}

    
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.swift
// import Foundation

// func solution(_ n:Int, _ a:Int, _ b:Int) -> Int
// {
//     var answer = 1
    
//     var A = min(a,b)
//     var B = max(a,b)
    
//     while (A,B) != (1,2) {
//         answer += 1
//         if (A % 2 == 0) {
//             A /=  2
//         } else {
//             A = (A / 2) + 1
//         }
        
//         if (B % 2 == 0) {
//             B /=  2
//         } else {
//             B = (B / 2) + 1
//         }
//     }
    
//     return answer
// }


import Foundation

func solution(_ n: Int, _ a: Int, _ b: Int) -> Int {
    var A = a
    var B = b
    var round = 0

    while A != B {
        A = (A + 1) / 2
        B = (B + 1) / 2
        round += 1
    }

    return round
}
